src/psyclone/tests/psyir/tools/definition_use_chains_backward_dependence_test.py

```diff
@@ -742,3 +742,34 @@ def test_definition_use_chains_backward_accesses_inquiry_func(
     )
     reaches = chains.find_backward_accesses()
     assert len(reaches) == 0
+
+
+def test_definition_use_chain_find_backward_accesses_pure_call(
+    fortran_reader,
+):
+    """Functionality test for the find_backward_accesses routine. This
+    tests the behaviour for a pure subrotuine call."""
+    code = """
+    pure subroutine y(in)
+        integer :: in
+        in = in + 1
+    end subroutine y
+
+    subroutine x(a, b)
+    integer :: a, b
+    a = 2
+    call y(b)
+    a = a + 2
+    b = 1
+    end subroutine"""
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Routine)[1]
+    chains = DefinitionUseChain(routine.children[2].rhs.children[0])
```

Comment:
More robust (and readable) to do a `walk`, e.g.
```python
assign = routine.walk(Assignment)[1]
chains = DefinitionUseChain(assign.rhs.children[0])
```

Reply:
Done

Reply:
Great, that's much easier on my brain, thanks :-)